indexing-service/src/main/java/io/druid/indexing/common/actions/TaskActionToolbox.java

```diff
@@ -126,4 +131,31 @@ public boolean apply(TaskLock taskLock)
 
     return true;
   }
+
+  // Verify that the segment intervals are covered by lock which is non-preemptive
+  public boolean locksAreNonPreemptive (Task task, Set<DataSegment> segments) {
+    // Get the locks from TaskLockbox (at overlord)
+    final List<TaskLock> taskLocks = getTaskLockbox().findLocksForTask(task);
+    // Each segment should be covered by at least a lock which is non-preemptive
+    for (final DataSegment segment : segments) {
+      final boolean isNonPreemptive = Iterables.all(
```

Comment:
[non-blocking comment] this could probably be a little clearer without so many double-negatives